### PR TITLE
🌱 Backport: Add unit test to m3 machine template controller

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
Backporting following commits to release-0.5:

1. https://github.com/metal3-io/cluster-api-provider-metal3/pull/446   
2. https://github.com/metal3-io/cluster-api-provider-metal3/pull/540

### CODE REVIEW

**Comment by fmuyassarov** [comment]
/retitle 🌱 Backport: Add unit test to m3 machine template controller

**Comment by fmuyassarov** [comment]
/test-v1a5-ubuntu-integration
/approve
Thank you @adilGhaffarDev

File: Makefile [hunk 1]
```diff
@@ -280,6 +287,11 @@ generate-go: $(CONTROLLER_GEN) $(MOCKGEN) $(CONVERSION_GEN) $(KUBEBUILDER) $(KUS
 		--output-file-base=zz_generated.conversion  $(CONVERSION_GEN_OUTPUT_BASE) \
 		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt
 
+	$(CONVERSION_GEN) \
+		--input-dirs=./api/v1alpha5 \
+		--output-file-base=zz_generated.conversion  $(CONVERSION_GEN_OUTPUT_BASE) \
+		--go-header-file=./hack/boilerplate/boilerplate.generatego.txt
+
```
Comment (furkatgofurov7) [inline Makefile:294]: Why do we need this conversion when we are already in the release-0.5 branch? 
In main we [do](https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/Makefile#L318-L321) conversion gen to v1alpha5 (we are in v1beta1), in release-0.4 branch we [do](https://github.com/metal3-io/cluster-api-provider-metal3/blob/release-0.4/Makefile#L267-L270) conversion gen to v1alpha3 (we are in v1alpha4). But here we are in release-0.5 branch already and doing conversion to v1alpha5?

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (adilGhaffarDev) [reply Makefile:294]: Yes you are right, It should be conversion gen to v1alpha4 and conversion to v1alpha5 should not be there. I am removing it.

**Review by furkatgofurov7 — APPROVED** [review]
/approve

**Review by furkatgofurov7 — COMMENTED** [review]
/lgtm

**Pull request merged**